### DISCUSSION TITLE
Cypress Gateway Connector requirement?

### DISCUSSION THREAD

**Discussion by mogusbi-motech** [sorry-cypress/sorry-cypress#787] [General]
- [x] I have searched issues for similar feature requests

## What is this feature about (1 sentence)

I have received an email from Cypress telling me I need to implement [Cypress Gateway Connector](https://go.cypress.io/continue-recording-with-gateway-connector?id=) to continue using `sorry-cypress`.

## Why is it needed? What is the value? For whom do we build it?

Not sure it is. Wanting some clarity as to whether I should ignore the warnings.

## Other

Here's the email I received from someone at Cypress

> We recently released changes to encrypt our private Cypress API requests for improving network security and stability of the overall Cypress testing experience. We are currently rolling this out across our user base and expect this to be complete effective June 30, 2023.
> 
> We have identified that this might impact the recording of tests for your projects using an on-premise instance of Sorry Cypress.
> 
> You will not be obligated to migrate off Sorry Cypress, however your action will be required to continue recording without disruption.

**Comment by agoldis** [comment]
tl;dr

Start using https://github.com/currents-dev/cypress-cloud package, it doesn't use any private API and is not subject to Cypress team's effort to block sorry-cypress

---

Starting from version 11.0.0 (November 2022), the [Cypress.io](http://cypress.io/) team started a series of released aimed to block sorry-cypress users.  First, they changed the internal structure of cypress package:

- the internal modules are now delivered as an [Electron binary snapshot](https://currents.dev/readme/guides/cypress-compatibility/cypress-11#cypress-11-and-v8-snapshots) that loads directly into memory to speed up the startup time
- the entry point is now a pre-compiled serialized v8 bytecode file, that has a series of [integrity checks](https://github.com/cypress-io/cypress/blob/8888cd9e211e608f9bbac81478667b2877dab76e/scripts/binary/binary-integrity-check-source.js) aimed to detect changes in the key package files, as well as the runtime environment

That prevented sorry-cypress from just changing the internal configuration. We managed to overcome those technical measures by rewriting the integration and using a network proxy for rerouting the orchestration requests.

We noticed that Cypress.io team continued putting limitations on the runner. Starting from version 12.7.0 (Feb 2023) the test runner [introduced](https://github.com/cypress-io/cypress/pull/25837) a new technical challenge - it collects (with no consent) information about the runtime environment, including:

- OS [process tree](https://github.com/cypress-io/cypress/blob/45c8a33f1c3e8d62ed419181a61a63e0b98d3543/packages/server/lib/cloud/environment.ts#L10) and detecting the executable names
- [environment variables](https://github.com/cypress-io/cypress/pull/25837/files#diff-e37e3536a71e33eaca58c7239d2976fbd90442059d6f54f2c648709cf38cc58dR51)
- [npm dependencies](https://github.com/cypress-io/cypress/blob/45c8a33f1c3e8d62ed419181a61a63e0b98d3543/packages/server/lib/cloud/environment.ts#L121) installed within the project

The information is transmitted to a backend service in an encoded and signed format as a [preflight](https://github.com/cypress-io/cypress/blob/cd7da984bc52df6e22d8fc30a5666b700504201d/packages/server/lib/cloud/api.ts#L290) request. Depending on the response from the backend, the runner may stop the execution with an error message 

```tsx
Cypress does not support recording test results to this third-party service.
```

In addition, Cypress team has been actively working on moving the source code to a closed-source repository, they want to limit the functionality available in their open-source version: https://github.com/cypress-io/cypress/pull/26188

In May 2023 they started reaching out to Sorry Cypress users, offering a "Cypress Gateway Connector" for a $US500/mo "subscription" fee. Enabling the feature would remove paying users from the block list.

---

https://github.com/currents-dev/cypress-cloud

We have released a new integration, with orchestration and reporting implemented from scratch and are now independent of cypress private APIs. The integration is available as https://github.com/currents-dev/cypress-cloud package.  It uses the public APIs and runs cypress in an “offline” mode as a standalone runner. Bottom line - we are not interfering with cypress internals anymore, we are using all the same APIs that are available for every developer and that are being used by hundreds of [plugins and complimentary tools](https://docs.cypress.io/plugins) of Cypress ecosystem.

**Reply by mirobo** [reply]
> In addition, Cypress team has been actively working on moving the source code to a closed-source repository, they want to limit the functionality available in their open-source version: https://github.com/cypress-io/cypress/pull/26188

@agoldis Where do you draw this information / hint from that they're trying to move to a closed-source repo? Is it this protocol manager?

**Reply by agoldis** [reply]
I think so

**Reply by fleks-dev** [reply]
Is there a free self-hosted version of currents.dev?

**Reply by samixchoumi** [reply]
@ktxxt yes, sorry-cypress

**Reply by fleks-dev** [reply]
Which will stop working from 30. June, if I'm not wrong?

Or you want to tell me that I can keep continue using Sorry-Cypress Dashboard but I have to switch from `cy2` to  `https://github.com/currents-dev/cypress-cloud` ?

**Reply by mathpaquette** [reply]
> Which will stop working from 30. June, if I'm not wrong?
> 
> 
> 
> Or you want to tell me that I can keep continue using Sorry-Cypress Dashboard but I have to switch from `cy2` to  `https://github.com/currents-dev/cypress-cloud` ?

@agoldis Yeah, I'm not clear on that one as well... could you please clarify?

**Reply by agoldis** [reply]
@mathpaquette 

Start using https://github.com/currents-dev/cypress-cloud package, it doesn't use any private API and is not subject to Cypress team's effort to block sorry-cypress

**Reply by xMH85x** [reply]
Well there is a problem with that plugin. If I use it locally it works, but if I want to use it in Teamcity, I get an error message: „cannot reading branch“. Teamcity blocks that information. The cy2 plugin works without those information about VCS. Is there a way to use that plugin without ans VCS information?

**Comment by samixchoumi** [comment]
Exactly
